packages/lib/hooks/useNotifications.tsx

```diff
@@ -0,0 +1,155 @@
+
+import { trpc } from "@calcom/trpc/react";
+import { showToast } from "@calcom/ui";
+
+export const useNotifications = () => {
+  const [buttonToShow, setButtonToShow] = useState<"none" | "allow" | "disable" | "denied">("none");
+  const [isLoading, setIsLoading] = useState(false);
+
+  const { mutate: addSubscription } = trpc.viewer.addNotificationsSubscription.useMutation({
+    onSuccess: () => {
+      setButtonToShow("disable");
+      showToast("Notifications turned on", "success");
+    },
+    onError: (error) => {
+      showToast(`Error: ${error.message}`, "error");
+    },
+    onSettled: () => {
+      setIsLoading(false);
+    },
+  });
+  const { mutate: removeSubscription } = trpc.viewer.removeNotificationsSubscription.useMutation({
+    onSuccess: () => {
+      setButtonToShow("allow");
+      showToast("Notifications turned off", "success");
+    },
+    onError: (error) => {
+      showToast(`Error: ${error.message}`, "error");
+    },
+    onSettled: () => {
+      setIsLoading(false);
+    },
+  });
+
+  useEffect(() => {
+    const decideButtonToShow = async () => {
+      if (!("Notification" in window)) {
+        console.log("Notifications not supported");
+      }
+
+      const registration = await navigator.serviceWorker.getRegistration();
+      if (!registration) return;
+      const subscription = await registration.pushManager.getSubscription();
+
+      const permission = Notification.permission;
+
+      if (permission === "denied") {
+        setButtonToShow("denied");
+        return;
+      }
+
+      if (permission === "default") {
+        setButtonToShow("allow");
+        return;
+      }
+
+      if (!subscription) {
+        setButtonToShow("allow");
+        return;
+      }
+
+      setButtonToShow("disable");
+    };
+
+    decideButtonToShow();
+  }, []);
+
+  const enableNotifications = async () => {
+    setIsLoading(true);
+    const permissionResponse = await Notification.requestPermission();
+
+    if (permissionResponse === "denied") {
+      setButtonToShow("denied");
+      setIsLoading(false);
+      showToast("You denied the notifications", "warning");
+      return;
+    }
+
+    if (permissionResponse === "default") {
+      setIsLoading(false);
+      showToast("Please allow notifications from the prompt", "warning");
+      return;
+    }
+
+    const registration = await navigator.serviceWorker.getRegistration();
+    if (!registration) {
+      // This will not happen ideally as the button will not be shown if the service worker is not registered
+      return;
+    }
+
+    let subscription: PushSubscription;
+    try {
+      subscription = await registration.pushManager.subscribe({
+        userVisibleOnly: true,
+        applicationServerKey: urlB64ToUint8Array(process.env.NEXT_PUBLIC_VAPID_PUBLIC_KEY || ""),
+      });
+    } catch (error) {
+      // This happens in Brave browser as it does not have a push service
+      console.error(error);
```

Comment:
you can enable push notifications on brave by enabling this and relaunching browser

![Screenshot 2024-08-22 at 1 35 33 PM](https://github.com/user-attachments/assets/d947cdb4-8e15-4d6f-8a63-2e17a144981c)

Reply:
Yes, so I was wondering if there is some browser API that we can call on the user's confirmation to enable this for them on the brave browser. What do you say?

Reply:
I saw there is no such option, but should we guide user by showing screenshot? and probably redirect them this setting?

Reply:
brave://settings/?search=push+messaging

You can add this link and it would redirect to settings page with option highlighted. No need for screenshot.